cpp/arcticdb/storage/library_manager.cpp

```diff
@@ -158,7 +158,10 @@ std::shared_ptr<Library> LibraryManager::get_library(const LibraryPath& path, co
 
 void LibraryManager::close_library_if_open(const LibraryPath &path) {
     std::lock_guard<std::mutex> lock{open_libraries_mutex_};
-    open_libraries_.erase(path);
+    if (auto it = open_libraries_.find(path); it != open_libraries_.end()) {
+        it->second->close();
```

Comment:
The naming is a bit weird, you have `close()` and `close_library_if_open` but these are both actually deleting files. Think a bit of renaming would be good.

Reply:
I think `cleanup_library_if_open` might be a better name. Similarly, in storage it would be called `cleanup()` rather than `close()`. I am not calling it `delete_library` or similar for storages because it doesn't do anything for all the other storages except lmdb